server/src/models/Trail.js

```diff
@@ -5,6 +5,34 @@ class Trail extends Model {
     return "trails"
   }
 
+  static get relationMappings() {
+    const { Review, User } = require("./index")
+
+    return {
+      reviews: {
+        relation: Model.HasManyRelation,
+        modelClass: Review,
+        join: {
+          from: "trails.id",
+          to: "reviews.trailId",
+        },
+      },
+
+      users: {
```

Comment:
What is your join table/ association between a Trail and User for needing a many-to-many?

A User has many Trails
A Trail belongs to a User

Reply:
Please refer to the ER diagram provided - a user can have many reviews of a trail and trail can have many reviews from a user.

Reply:
When querying the trail - the instance of a trail via the relational mapping will allow us to see the users

Reply:
When querying the trail - the instance of a trail via the relational mapping will allow us to see the users who have reviewed the trail.

Reply:
This `users` association is defining the relationship between a Trail and a User, which a one-to-many / belongs-to. If you look at your ERD, the relation between a trail and user is:
A User has many Trails
A Trail belongs to a User

It is still true that a User can have many reviews, and those reviews can be made on the same Trail. If you want the User who left a Review for a Trail, you would need to go through your associations within serializers. From the Trail, you can find all of its Reviews, then from those Reviews you can find the User who made that review.

Reply:
Sorry - We do not understand this - really confused.